packages/service-library/requirements/_fastapi.in

```diff
@@ -9,6 +9,7 @@
 async-asgi-testclient # replacement for fastapi.testclient.TestClient [see b) below]
 fastapi
 fastapi_contrib[jaegertracing]
+httpx
```

Comment:
This is a requirement for the module. Not for testing.

Reply:
Why is a testing dependency inside this file though? Should it not go in the `_tests.in`?
I will see if the LifeCycle events work with the `async-asgi-testclient`. I did not notice it.

Reply:
I just remembered why this is like this. When testing the `client` part, an instance of `httpx.AsyncClient` is required by design.
This also got reused for testing the other parts of the application.

Regarding the `asgi_lifespan` is required for the server. In the `test_long_running_context_manager.py` what you suggest will not work. Since the client and server side are meant to be separate. Ideally they will be in separate processes. But for the sake of testing they stay in the same loop here.